tmt/steps/report/reportportal.py

```diff
@@ -607,19 +622,15 @@ def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
                             response = self.rp_api_post(
                                 session=session,
                                 path="log/entry",
-                                json={"message": _filter_invalid_chars(
-                                    _filter_log_per_size(log,
-                                                         int(self.data.log_size_limit))),
+                                json={"message": self.data._filter_log(log),
```

Comment:
Hm, probably I got confused by your proposal to use `ReportReportPortalData` type in an argument. I though that by bonding it with this specific class including it inside of it would just simplify data access. But overall I agree it is not that much related, but I would personally avoid using ReportReportPortalData type argument completely and revert back to the `limit` argument. OTOH, that was probably a way how to overcome the varying number of arguments in filter functions, correct?

Reply:
Exactly, it was a way how to reduce both (and future) filter callable to the same denominator while giving them access to phase parameters.

Reply:
Updated.